src/DocumentFormat.OpenXml/OpenXmlReader.cs

```diff
@@ -241,6 +241,47 @@ public virtual bool HasAttributes
         /// <remarks>The new current element is the end of the element after LoadCurrentElement().</remarks>
         public abstract OpenXmlElement LoadCurrentElement();
 
+        /// <summary>
+        /// Loads the element at current cursor as the specified type.
+        /// </summary>
+        /// <typeparam name="T">The concrete type of the OpenXmlElement to load.</typeparam>
+        /// <returns>The OpenXmlElement that was loaded.</returns>
+        /// <exception cref="InvalidOperationException">Thrown when the current is the end element or when the ElementType and <typeparamref name="T"/> do not match.</exception>
+        public virtual T LoadElement<T>()
+            where T : OpenXmlElement
+        {
+            if (ElementType == typeof(T))
+            {
+                return (T)LoadCurrentElement();
+            }
+            else
+            {
+                throw new InvalidOperationException(ExceptionMessages.ReaderIncorrectTypeOnLoad);
+            }
+        }
+
+        /// <summary>
+        /// Attempts to load the element at current cursor as the specified type.
+        /// </summary>
+        /// <typeparam name="T">The concrete type of the OpenXmlElement to load.</typeparam>
+        /// <param name="element">The OpenXmlElement that was loaded.</param>
+        /// <returns><c>true</c> if the element was loaded, otherwise <c>false</c>.</returns>
+        /// <exception cref="InvalidOperationException">Thrown when the current is the end element.</exception>
```

Comment:
Can you check to see if it is the end element so we don't throw here?

Reply:
i feel that it still should throw because if this is in a loop, it should behave the same way.

Reply:
I don't have a strong opinion on this behavior, so as along as it is documented and a test to verify the behavior, I'm OK with that.